cli/internal/env/env.go

```diff
@@ -56,6 +56,10 @@ type EnvironmentVariablePairs []string
 // mapToPair returns a deterministically sorted set of EnvironmentVariablePairs from an EnvironmentVariableMap
 // It takes a transformer value to operate on each key-value pair and return a string
 func (evm EnvironmentVariableMap) mapToPair(transformer func(k string, v string) string) EnvironmentVariablePairs {
+	if evm == nil {
+		return nil
+	}
```

Comment:
For my understanding, how would this ever be nil? Do the tests fail without this check?

Reply:
You can manually create a nil `EnvironmentVariableMap`. This prevents transformation of `nil` to `[]`. And yes, tests fail without this change.